components/apps.js

```diff
@@ -649,33 +647,137 @@ SteamUser.prototype.getOwnedDepots = function(excludeSharedLicenses) {
+		return packages.filter(license => license.flags !== SteamUser.ELicenseFlags.Expired).map(license => license.package_id);
 	}
+	// If options is a boolean, we asssume it's excludeSharedLicenses
+	if (typeof options === 'boolean') {
+		let excludeSharedLicenses = options;
+		if (excludeSharedLicenses) {
+			packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+		}
+		// Slight deviation of old method: don't return expired licenses
+		return packages.filter(license => license.flags !== SteamUser.ELicenseFlags.Expired).map(license => license.package_id);
+	}
+
+	// Handle options
+	if (typeof options === 'object') {
+		const defaults = {
+			excludeFree: false, // By default, include free licenses (Sub 0 & FreeOnDemand & NoCost)
+			excludeShared: true, // By default, exclude shared licenses
+			excludeExpired: true // By default, exclude licenses that are going to expire (free weekends)
+		};
+		options = Object.assign(defaults, options);
+
+		// If there is nothing to filter, no pics cache needed
+		if (!options.excludeFree && !options.exludeShared && !options.excludeFree) {
+			return packages.map(license => license.package_id);
+		}
+	}
+
+	// From this point, we need pics cache to be enabled
+	this._ensurePicsCache();
+
+	// Determine the filter function
+	let packageFilter = () => true;
+	if (typeof options === 'function') {
+		packageFilter = options;
+	} else if (typeof options === 'object') {
+		packageFilter = (license) => {
+
+			// If expired, filter it out, regardless of the options
+			if (license.flags === SteamUser.ELicenseFlags.Expired) {
+				return false;
+			}
+
+			// If exclude shared licenses
+			if (options.excludeShared && license.owner_id !== this.steamID.accountid) {
+				return false;
+			}
+
+			let pkg = license.package_id;
+			if (!this.picsCache.packages[pkg]) {
+				this._warn(`Failed to filter package ${pkg.packageid} (no pics cache info available)`);
+				return false;
+			}
+
+			pkg = this.picsCache.packages[pkg];
+			if (!pkg.packageinfo) {
+				this._warn(`Failed to filter package ${pkg.packageid} (no pics cache info available)`);
+				return false;
+			}
+
+			pkg = pkg.packageinfo;
+			let owned = true;
+
+			// If exclude all free (sub 0, FreeOnDemand, or NoCost)
+			if (options.excludeFree) {
+				owned = owned
+						&& pkg.packageid !== 0
+						&& pkg.billingtype !== SteamUser.EBillingType.NoCost
+						&& pkg.billingtype !== SteamUser.EBillingType.GuestPass
+						&& pkg.billingtype !== SteamUser.EBillingType.FreeOnDemand
+						&& pkg.billingtype !== SteamUser.EBillingType.FreeCommercialLicense;
+			}
+
+			// If not temporary (free promotions are yours to keep permanently)
+			if (!pkg.extended || !pkg.extended.expirytime || pkg.extended.freepromotion) {
+				return owned;
+			}
+
+			// If exclude all expiring licenses
+			if (options.excludeExpired) {
+				return false; // return false, since this license is temporary (does not matter if not expired yet)
```

Comment:
I don't really like the idea of excluding expiring-but-not-yet-expired along with expired licenses.